model-optimizer/extensions/front/RollWithEmptyAxesReplacer.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from extensions.front.AttributedRollToRoll import AttributedRollToRoll
+from extensions.ops.roll import Roll
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementOp
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph, rename_nodes
+from mo.ops.const import Const
+from mo.ops.reshape import Reshape
+from mo.ops.shape import Shape
+
+
+class RollWithEmptyAxesReplacer(FrontReplacementOp):
+    """
+    This transformation replaces Roll with empty axes input with the following sequence of operations:
+    reshape to 1D tensor -> roll -> reshape to original shape.
+    """
+    enabled = True
+    op = "Roll"
+
+    def run_after(self):
+        return [AttributedRollToRoll]
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        roll_node = match['op']
+        if not roll_node.in_port(2).disconnected():
+            return
+        node_name = roll_node.soft_get('name', roll_node.id)
+
+        # reshape to 1d tensor
+        reshape_to_1d = create_op_node_with_second_input(graph, Reshape, int64_array([-1]),
+                                                         {'name': node_name + '/reshape'})
+        roll_node.in_port(0).get_connection().set_destination(reshape_to_1d.in_port(0))
+
+        # roll
+        const_zero = Const(graph, {'value': int64_array([0]), 'name': node_name + '/axes'}).create_node()
+        new_roll = Roll(graph, {'name': node_name + '/roll'}).create_node()
```

Comment:
We don't need to create a new instance of Roll operation. Instead, insert the Reshape operation before and after the Roll like this:
```python
roll_node.in_port(0).get_connection().insert_node(reshape_to_1d)
roll_node.out_port(0).get_connection().insert_node(reshape_to_orig_shape)
```